src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -1128,6 +1140,345 @@ subroutine iceshelf_melt_ismip6(domain, err)
+         ! Get ISMIP6 forcing fields
+         call mpas_pool_get_array(geometryPool, 'ismip6_2dThermalForcing', TFocean) 
+         call mpas_pool_get_array(geometryPool, 'ismip6Runoff', ismip6Runoff)
+         call mpas_pool_get_array(geometryPool, 'calvingVelocity', calvingVelocity)
+        
+         ! Get mesh and geometry arrays
+         call mpas_pool_get_array(geometryPool, 'cellMask', cellMask)
+         call mpas_pool_get_array(meshPool, 'areaCell', areaCell)
+         call mpas_pool_get_array(meshPool, 'dvEdge', dvEdge)
+         call mpas_pool_get_array(meshPool, 'areaCell', areaCell)
+         call mpas_pool_get_array(meshPool, 'nEdgesOnCell', nEdgesOnCell)
+         call mpas_pool_get_array(meshPool, 'cellsOnCell', cellsOnCell)
+         call mpas_pool_get_array(meshPool, 'edgesOnCell', edgesOnCell)
+         call mpas_pool_get_array(meshPool, 'cellsOnEdge', cellsOnEdge)
+         call mpas_pool_get_array(geometryPool, 'thickness', thickness)
+         call mpas_pool_get_array(geometryPool, 'meltingThickness', meltingThickness)
+         call mpas_pool_get_array(geometryPool, 'bedTopography', bedTopography)
+         call mpas_pool_get_array(geometryPool, 'lowerSurface', lowerSurface) 
+         call mpas_pool_get_array(geometryPool, 'groundedMarineMarginMask', groundedMarineMarginMask)
+         call mpas_pool_get_array(geometryPool, 'faceMeltRateApplied', faceMeltRateApplied)
+         call mpas_pool_get_array(geometryPool, 'faceMeltSpeed', faceMeltSpeed)
+         
+       ! Calculate retreat rate for each cell
+         do iCell = 1, nCellsSolve
+            ! Define marine marginal cells as those that are (1) at the ice
+            ! margin, (2) have at least one neighboring cell without ice, (3) contain
+            ! grounded ice, and (4) have bed topography below sea level.
+            ! OR is adjacent to an inactive floating margin cell
+            nEmptyNeighbors = 0
+            oceanEdgeLength = 0.0_RKIND
+            ! Check if neighboring cells contain ice and have bed topo below sea
+            ! level
+            do iEdge = 1, nEdgesOnCell(iCell)
+               iNeighbor = cellsOnCell(iEdge, iCell)
+                 if ( ((thickness(iNeighbor) == 0.0_RKIND) & 
+                    .and. iNeighbor > 0 & ! make sure this cell exists
+                    .and. iNeighbor <= nCellsSolve & ! make sure this cell exists
+                    .and. bedTopography(iNeighbor) < seaLevel) &
+                    .or. (li_mask_is_floating_ice(cellMask(iNeighbor)) & 
+                    .and. li_mask_is_margin(cellMask(iNeighbor)) &
+                    .and. (.not. li_mask_is_albany_active(cellMask(iNeighbor)))) ) then
+                     nEmptyNeighbors = nEmptyNeighbors + 1
+                     oceanEdgeLength = oceanEdgeLength + dvEdge(edgesOnCell(iEdge, iCell))
+                 endif
+            enddo 
+            
+            if ( nEmptyNeighbors > 0 &
+               .and. thickness(iCell) > 0.0_RKIND &
```

Comment:
No need to check if thickness>0 if you are already checking the li_mask_is_grounded_ice mask value.